vault/programs/vault/src/context.rs

```diff
@@ -0,0 +1,323 @@
+}
+
+#[derive(Accounts)]
+pub struct InitializeZetaMarginAccount<'info> {
+    pub zeta_program: AccountInfo<'info>,
+    pub vault_authority: Signer<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_authority,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    pub initialize_margin_cpi_accounts: InitializeMarginAccount<'info>,
+}
+
+#[derive(Accounts)]
+pub struct DepositZeta<'info> {
+    pub zeta_program: AccountInfo<'info>,
+    pub vault_authority: Signer<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_authority,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    pub deposit_cpi_accounts: Deposit<'info>,
+}
+
+#[derive(Accounts)]
+pub struct WithdrawZeta<'info> {
+    pub zeta_program: AccountInfo<'info>,
+    pub vault_authority: Signer<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_authority,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    pub withdraw_cpi_accounts: Withdraw<'info>,
+}
+
+#[derive(Accounts)]
+pub struct InitializeZetaOpenOrders<'info> {
+    pub zeta_program: AccountInfo<'info>,
+    pub vault_authority: Signer<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_authority,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    pub initialize_open_orders_cpi_accounts: InitializeOpenOrders<'info>,
+}
+
+#[derive(Accounts)]
+pub struct PlaceAuctionOrder<'info> {
+    pub zeta_program: AccountInfo<'info>,
+    pub vault_authority: Signer<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_authority,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    pub place_order_cpi_accounts: PlaceOrder<'info>,
+}
+
+#[derive(Accounts)]
+pub struct CancelAuctionOrder<'info> {
+    pub zeta_program: AccountInfo<'info>,
+    pub vault_authority: Signer<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_authority,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    pub cancel_order_cpi_accounts: CancelOrder<'info>,
+}
+
+#[derive(Accounts)]
+#[instruction(vault_name: String, bumps: VaultBumps)]
+pub struct RolloverVault<'info> {
+    // vault Authority accounts
+    #[account(mut)]
+    pub vault_authority: Signer<'info>,
```

Comment:
Add constraint to check that vault_authority == vault.vault_authority, otherwise anyone can call this

Reply:
Good pickup